src/Microsoft.AspNetCore.OData/Extensions/Endpoint/ODataEndpointSelector.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+#if !NETSTANDARD2_0
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNet.OData.Interfaces;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Routing.Matching;
+
+namespace Microsoft.AspNet.OData.Extensions
+{
+    /// <summary>
+    /// A service that is responsible for the final OData Endpoint selection decision.
+    /// This selector is used to resolve the duplicated Endpoint selection.
+    /// </summary>
+    internal class ODataEndpointSelector : EndpointSelector
+    {
+        private EndpointSelector _innerSelector;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ODataEndpointSelector"/> class.
+        /// </summary>
+        /// <param name="selector">The inner Endpoint selector</param>
+        public ODataEndpointSelector(EndpointSelector selector)
+        {
+            _innerSelector = selector;
+        }
+
+        /// <inheritdoc/>
+        public override Task SelectAsync(HttpContext httpContext, CandidateSet candidateSet)
+        {
```

Comment:
What are the other things that can happen in the pipeline while this runs asynchronously?